
This enables:

-PCIe

Author: Piotr Oniszczuk <piotr.oniszczuk@gmail.com>
Based on work of:
   Michael Riesch <michael.riesch@wolfvision.net>
   Frank Wunderlich <frank-w@public-files.de>
   Peter Geis <pgwipeout@gmail.com>


diff --speed-large-files --no-dereference --minimal -Naur linux-5.18-rc6/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts linux-5.18-rc6/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts
--- linux-5.18-rc6/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2022-05-10 16:27:41.793187015 +0200
+++ linux-5.18-rc6/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2022-05-10 17:00:22.993176106 +0200
@@ -1,5 +1,13 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR MIT)
 
+/*
+ * Author: Piotr Oniszczuk <piotr.oniszczuk@gmail.com>
+ * Based on work of:
+ *   Michael Riesch <michael.riesch@wolfvision.net>
+ *   Frank Wunderlich <frank-w@public-files.de>
+ *   Peter Geis <pgwipeout@gmail.com>
+ */
+
 /dts-v1/;
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
@@ -128,6 +136,38 @@
 		regulator-name = "vcc5v0_hub";
 		regulator-always-on;
 	};
+
+	pcie30_avdd0v9: pcie30-avdd0v9 {
+		compatible = "regulator-fixed";
+		regulator-name = "pcie30_avdd0v9";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <900000>;
+		regulator-max-microvolt = <900000>;
+		vin-supply = <&vcc3v3_sys>;
+	};
+
+	pcie30_avdd1v8: pcie30-avdd1v8 {
+		compatible = "regulator-fixed";
+		regulator-name = "pcie30_avdd1v8";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		vin-supply = <&vcc3v3_sys>;
+	};
+
+	vcc3v3_pcie_p: vcc3v3-pcie-p {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpio = <&gpio0 RK_PD4 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pcie_enable_h>;
+		regulator-name = "vcc3v3_pcie_p";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&vcc12v_dcin>;
+	};
 };
 
 &cpu0 {
@@ -464,6 +504,26 @@
 	status = "okay";
 };
 
+&pcie30phy {
+	status = "okay";
+};
+
+&pcie3x2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie_reset_x2>;
+	reset-gpios = <&gpio2 RK_PD6 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie_p>;
+	status = "okay";
+};
+
+&pcie2x1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie_reset_x1>;
+	reset-gpios = <&gpio3 RK_PC1 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie_p>;
+	status = "okay";
+};
+
 &pinctrl {
 	ethernet {
 		eth_phy_rst: eth_phy_rst {
@@ -497,6 +557,21 @@
 			rockchip,pins = <0 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	pcie {
+		pcie_enable_h: pcie-enable-h {
+			rockchip,pins = <0 RK_PD4 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+
+		pcie_reset_x1: pcie-reset-x1 {
+			rockchip,pins = <3 RK_PC1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		pcie_reset_x2: pcie-reset-x2 {
+			rockchip,pins = <2 RK_PD6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };
 
 &pmu_io_domains {
